src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/CacheTagKey.cs

```diff
@@ -87,11 +91,17 @@ private CacheTagKey(CacheTagHelperBase tagHelper)
             _queries = ExtractCollection(tagHelper.VaryByQuery, request.Query, QueryAccessor);
             _routeValues = ExtractCollection(tagHelper.VaryByRoute, tagHelper.ViewContext.RouteData.Values, RouteValueAccessor);
             _varyByUser = tagHelper.VaryByUser;
+            _varyByCulture = tagHelper.VaryByCulture;
 
             if (_varyByUser)
             {
                 _username = httpContext.User?.Identity?.Name;
             }
+
+            if (_varyByCulture)
+            {
+                _requestCulture = CultureInfo.CurrentCulture.Name;
```

Comment:
Combine both culture and uiculture names?

Reply:
I was going by the comment in the issue which suggested using `CultureInfo.CurrentCulture` - see https://github.com/aspnet/Mvc/issues/3938#issuecomment-172058243. @sebastienros should it be using both?